document/CoreFormat.mdk

```diff
@@ -340,6 +340,48 @@ run: {
 ```
 ~
 
+### `logicalLocations` property { #run-logicalLocations }
+
+A `run` object should contain a property named `logicalLocations` whose value is a dictionary,
+each of whose properties represents the logical location of one or more results produced in the course of the run.
+
+With one exception described in &sect;[#location-logicalLocationKey], each property name in the `logicalLocations` object shall be
+a string representing the logical location where the result was produced, in a format consistent with
+the programming language in which the programmatic construct specified by that logical location was expressed.
+We refer to this string as a "fully qualified logical name".
+See &sect;[#location-fullyQualifiedLogicalName] for examples.
+
+Each value in the `logicalLocation` object shall be
+an array of one or more `logicalLocationComponent` objects (&sect;[#logicalLocationComponent]).
+This allows reference to results which occur in nested logical components, for example,
+a member nested within a type which is in turn nested within a namespace.
+When a result occurs in such a nested construct, the first `logicalLocationComponent` object in the array shall refer to
+the outermost construct,
+the next element shall refer to a construct nested immediately within the first construct,
+and so on, with the last element referring to the most deeply nested construct.
+
+~ IsoExample
+
+```
+"logicalLocations": {
+  "namespaceA::namespaceB::classC": [
+    {
+      "name": "namespaceA",
+      "kind": "namespace"
+    },
+    {
+      "name": "namespaceB",
```

Comment:
Can we remove the double-nesting of namespace here? this just seems to unnecessarily encourage bloating the log file,. i.e., doesn't buy us anything from simply expressing the namespace as a single item.